gnovm/pkg/gnolang/types.go

```diff
@@ -1033,22 +1040,26 @@ func (it *InterfaceType) IsImplementedBy(ot Type) (result bool) {
 			// ie, if each of ft's arg types can match
 			// against the desired arg types in im.Types.
 			if !gno2GoTypeMatches(im.Type, nft.Type) {
-				return false
+				return errors.New("types do not match")
 			}
 		} else if mt, ok := ft.(*FuncType); ok {
 			// if method is pointer receiver, check addressability:
 			if _, ptrRcvr := rt.(*PointerType); ptrRcvr && !hp {
-				return false // not addressable.
+				return fmt.Errorf(pointerReceiverMsg, im.Name) // not addressable.
 			}
 			// check for func type equality.
 			dmtid := mt.TypeID()
 			imtid := im.Type.TypeID()
 			if dmtid != imtid {
-				return false
+				return fmt.Errorf(wrongTypeMsg, im.Name)
 			}
 		}
 	}
-	return true
+	return nil
+}
+
+func (it *InterfaceType) IsImplementedBy(ot Type) bool {
```

Comment:
Is this used?

Reply:
hello thanks for the review yes it is used on the functions that used IsImplementedBy before
I decided to leave this function for the old functions that did not need any error returned for exemple:
```go
func IsImplementedBy(it Type, ot Type) bool {
	switch cbt := baseOf(it).(type) {
	case *InterfaceType:
		return cbt.IsImplementedBy(ot)
	case *NativeType:
		return gno2GoTypeMatches(ot, cbt.Type)
	default:
		panic("should not happen")
	}
}
```
this could also be expressed like
```go
		return cbt.IsImplementedBy(ot) == nil
```

But I didn't change a lot of code on this PR if you prefer I can remove this function and use the one with the returned error

Reply:
just checking, sounds good!